### PR TITLE
build: upgrade cargo crates and ctrl-plane

### PR DESCRIPTION
**Pull request by tiagolobocastro**
## Description
Upgrade control-plane dep and cargo crates

## Motivation and Context
Don't lag behind the control-plane.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.

### CODE REVIEW

**Review by Abhinandan-Purkait — APPROVED** [review]

**Review by niladrih — COMMENTED** [review]
Left one comment..

File: k8s/upgrade/Cargo.toml [hunk 1]
```diff
@@ -16,25 +16,25 @@ openapi = { path = "../../dependencies/control-plane/openapi", default-features
...
-kube = { version = "0.78.0",  default-features = true, features = [ "derive", "runtime" ] }
-anyhow = "1.0.72"
-clap = { version = "4.3.21", features = ["derive", "env", "string", "color"] }
+kube = { version = "0.85.0",  default-features = true, features = [ "derive", "runtime" ] }
+anyhow = "1.0.75"
+clap = { version = "4.4.6", features = ["derive", "env", "string", "color"] }
 humantime = "2.1.0"
 maplit = "1.0.2"
-k8s-openapi = { version = "0.17.0", features = ["v1_20"] }
+k8s-openapi = { version = "0.19.0", features = ["v1_20"] }
```
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply k8s/upgrade/Cargo.toml:24]: Not without testing that we're not using any feature which is no longer present on v1_20?
Also this change requires also changing control-plane repo beforehand to keep it in sync.

[the remark this answers is not in the document]
Comment (niladrih) [reply k8s/upgrade/Cargo.toml:24]: Sure, we can wait for the repo tests to get in before we make it.

I'll make sure to keep an eye out for ctrl-plane changes when they happen.

[the remark this answers is not in the document]
Comment (niladrih) [reply k8s/upgrade/Cargo.toml:24]: Okay, got it. This is fine for now I guess.

**Pull request merged**